src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -930,6 +930,92 @@ void createAcspMembersDataWithDefaultProfileValues() throws DataException {
         verify(acspMembersService).create(any(AcspMembersSpec.class));
     }
 
+    @Test
+    void createAcspMembersDataWhenProfileIsNotNull() throws DataException {
+        AcspMembersSpec spec = new AcspMembersSpec();
+        spec.setUserId("userId");
+
+        AcspProfileData acspProfileData = new AcspProfileData("acspNumber");
+        AcspMembersData acspMembersData = new AcspMembersData("memberId", acspProfileData.getAcspNumber(), "userId", "active", "role");
+        var acspStatus = "active";
+        var acspType = "ltd";
+        var supervisoryBody = "financial-conduct-authority-fca";
+        var membershipDetails = "Membership ID: FCA654321";
+
+        AcspProfileSpec acspProfile = new AcspProfileSpec();
+        acspProfile.setStatus(acspStatus);
+        acspProfile.setType(acspType);
+        AmlSpec amlSpec = new AmlSpec();
+        amlSpec.setSupervisoryBody(supervisoryBody);
+        amlSpec.setMembershipDetails(membershipDetails);
+
+        acspProfile.setAmlDetails(Collections.singletonList(amlSpec));
+
+        spec.setAcspProfile(acspProfile);
+
+        when(acspProfileService.create(any(AcspProfileSpec.class))).thenReturn(acspProfileData);
+        when(acspMembersService.create(any(AcspMembersSpec.class))).thenReturn(acspMembersData);
+
+        AcspMembersData result = testDataService.createAcspMembersData(spec);
+
+        assertNotNull(result);
+        assertEquals("acspNumber", result.getAcspNumber());
+        assertNotNull(spec.getAcspProfile());
+        assertTrue(spec.getAcspProfile() instanceof AcspProfileSpec);
+        assertTrue(spec.getAcspProfile().getAmlDetails().getFirst() instanceof AmlSpec);
```

Comment:
I don't think we need these assertions. spec is the input and we are setting it in this test